nsqd/lookup.go

```diff
@@ -118,35 +133,68 @@ func (n *NSQD) lookupLoop() {
 			}
 		case val := <-n.notifyChan:
 			var cmd *nsq.Command
+			var err error
 			var branch string
-
-			switch val.(type) {
-			case *Channel:
-				// notify all nsqlookupds that a new channel exists, or that it's removed
-				branch = "channel"
-				channel := val.(*Channel)
-				if channel.Exiting() == true {
-					cmd = nsq.UnRegister(channel.topicName, channel.name)
-				} else {
-					cmd = nsq.Register(channel.topicName, channel.name)
-				}
-			case *Topic:
-				// notify all nsqlookupds that a new topic exists, or that it's removed
-				branch = "topic"
-				topic := val.(*Topic)
-				if topic.Exiting() == true {
-					cmd = nsq.UnRegister(topic.name, "")
-				} else {
-					cmd = nsq.Register(topic.name, "")
+			notifyCtx, ok := val.(notifyContext)
+			if ok {
```

Comment:
I realize it's theoretically possible for there to be a case where a type other than `notifyContext` is sent over the channel, but it's not _practically_ possible (we can verify this manually at all call sites).

Let's drop the indentation here and do this instead:

```go
if !ok {
    panic("non-notifyContext sent to notifyChan - should never happen")
}
```

Reply:
done.